lib/cstruct.mli

```diff
@@ -181,6 +181,16 @@ val of_string: ?allocator:(int -> t) -> string -> t
     with the underlying buffer allocated by [alloc]. If [allocator] is not
     provided, [create] is used. *)
 
+(** {2 Comparison } *)
+
+val equal : t -> t -> bool
+(** [equal t1 t2] is [true] iff [t1] and [t2] correspond to the same sequence of
+    bytes. *)
+
+val compare : t -> t -> int
+(** [compare t1 t2] is [-1] if [t1] precedes [t2] by lexicographical order, [1]
+    if [t2] precedes [t1], and [0] otherwise. *)
```

Comment:
Well it gives the order its meaning. If you postulate that `"ABC"` < `"BCD"`, but `"ABC"` > `"BC"`, it just becomes totally abstract.

OTOH the real reason for `compare` is to have a fast `equal`, `compare` just happens to pop out half the way... and I realized that `equal` should indeed short-circuit.